core/server/master/src/main/java/alluxio/master/file/meta/MountTable.java

```diff
@@ -90,14 +90,15 @@ public boolean hasNext() {
         if (mEntry != null) {
           return true;
         }
-        while (it.hasNext()) {
+        if (it.hasNext()) {
           mEntry = it.next();
-          // Do not journal the root mount point.
-          if (!mEntry.getKey().equals(ROOT)) {
-            return true;
-          } else {
+          // Skip the root mount point, which is considered a part of initial state, not journaled
```

Comment:
I've reverted the changes to `clear` so that it also knows that the mount point is special